src/checkout-buttons/strategies/paypalCommerce/paypalCommerce-button-strategy.ts

```diff
@@ -0,0 +1,82 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { INTERNAL_USE_ONLY } from '../../../common/http-request';
+import { ApproveDataOptions, PaypalCommerceScriptLoader } from '../../../payment/strategies/paypalCommerce';
+import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
+import CheckoutButtonStrategy from '../checkout-button-strategy';
+
+export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrategy {
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _formPoster: FormPoster,
+        private _requestSender: RequestSender
+    ) {}
+
+    initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethod(options.methodId);
+        const paypalOptions = options.paypalCommerce;
+
+        if (!paypalOptions || !paypalOptions.clientId) {
+            throw new InvalidArgumentError();
+        }
+
+        if (!paymentMethod || !paymentMethod.id) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._paypalScriptLoader.loadPaypalCommerce(paypalOptions.clientId)
+            .then(paypal => {
+                return paypal.Buttons({
+                    createOrder: () => this._setupPayment(),
+                    onApprove: data => this._tokenizePayment(paymentMethod.id, data),
+                }).render(`#${options.containerId}`);
+            });
+    }
+
+    deinitialize(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    private _setupPayment(): Promise<string> {
+
+        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
+            .then(state => {
+                const url = '/api/storefront/payment/paypalcommerce';
+                const headers = {
+                    'X-API-INTERNAL': INTERNAL_USE_ONLY,
+                    'Content-Type': 'application/x-www-form-urlencoded',
```

Comment:
I'm curious why the content-type is not `application/json`? This looks like a regular storefront API so the content type should just be JSON.

Reply:
Yes, but this api works for paypal express too. And request with application/json doesn't work